.github/workflows/tests.yml

```diff
@@ -12,40 +12,50 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          cache: npm
       - run: npm ci
       - run: npm run lint
 
   integration:
     runs-on: ubuntu-latest
+    needs: lint
     strategy:
       matrix:
         eslint: ["4.7", latest, next]
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          cache: npm
       - run: npm ci
       - run: ESLINT_VERSION=${{ matrix.eslint }} ./tools/integration-tests.bash
 
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    needs: lint
     strategy:
       matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+        node:
+          - 12
+          - 14
+          - 16
+          - 17
+        eslint:
+          - latest
+          - next
         include:
           - node: 10
             eslint: 7
-          - node: 12
```

Comment:
Maybe it's better to just keep this one and 14 as a single OS run instead of expanding the matrix out so much